lib/tests/formulae_dependents.rb

```diff
@@ -169,8 +169,16 @@ def install_dependent(dependent, testable_dependents, args:, build_from_source:
         test "brew", "linkage", "--test", dependent.full_name
 
         if testable_dependents.include? dependent
-          test "brew", "install", "--only-dependencies", "--include-test",
-               dependent.full_name
+          test "brew", "install", "--only-dependencies", "--include-test", dependent.full_name
+
+          dependent.deps.each do |dependency|
+            next if dependency.build?
+            next if dependency.to_formula.keg_only?
+            next if dependency.to_formula.linked_keg.exist?
```

Comment:
nit: this calls `dependency.to_formula` twice. we could store the formula as `dependency_f` for example to save some computation